src/DurableTask.SqlServer.AzureFunctions/SqlScaleMonitor.cs

```diff
@@ -19,14 +19,22 @@ class SqlScaleMonitor : IScaleMonitor<SqlScaleMetric>
         static readonly ScaleStatus NoScaleVote = new ScaleStatus { Vote = ScaleVote.None };
         static readonly ScaleStatus ScaleOutVote = new ScaleStatus { Vote = ScaleVote.ScaleOut };
 
-        readonly SqlOrchestrationService service;
+        readonly SqlMetricsProvider metricsProvider;
 
         int? previousWorkerCount = -1;
 
-        public SqlScaleMonitor(SqlOrchestrationService service, string taskHubName)
+        public SqlScaleMonitor(string taskHubName, SqlMetricsProvider sqlMetricsProvider)
         {
-            this.service = service ?? throw new ArgumentNullException(nameof(service));
-            this.Descriptor = new ScaleMonitorDescriptor($"DurableTask-SqlServer:{taskHubName ?? "default"}");
+            // Scalers in Durable Functions are shared for all functions in the same task hub.
+            // So instead of using a function ID, we use the task hub name as the basis for the descriptor ID.
+            string id = $"DurableTask-SqlServer:{taskHubName ?? "default"}";
```

Comment:
huh, I think we've been getting this wrong across the board. This approach - one monitor per taskhub instead of a monitor per functionID, seems like what we should have done all along. If your tests show this works, that's great. I just hope there's no hidden dependency on the funcitonID being reflected on the monitor's own ID. 
Taking note.

Reply:
The MSSQL backend has been doing this always, and it's something I discovered when looking closely at the Azure Storage scale logs in the past (though for some reasons I never got around to suggesting we fix it). Basically, the existing scaler logic is creating redundant scalers that are all doing the exact same thing with the exact same data - which still works but is wasteful and can be costly to the customer, who gets billed for all these storage transactions.